graphHttpServer/GraphHttpServer.cpp

```diff
@@ -0,0 +1,271 @@
+        ss << "{name:" <<  EscapeJson(property->name())
+          << ",type:" << EscapeJson(property->typeName())
+          << ",value:" << converter.json() << "}";
+
+        if (i + 1 < list->numProperty()) {
+            ss << ",";
+        }
+    }
+    ss << "]";
+
+    reply->text += ss.str();
+}
+
+void ServeNode(NodeBase *node, HttpReply *reply) {
+    if (!node) {
+        reply->setNotFound();
+        return;
+    }
+
+    std::ostringstream ss;
+    ss << "{name:" << EscapeJson(node->name()) << ",output:[";
+
+    for (int i = 0; i < node->numOutputStream(); ++i) {
+        NamedStream *stream = node->outputStream(i);
+        ss << "{name:" << EscapeJson(stream->streamName())
+          << ",type:" << EscapeJson(stream->typeName()) << "}";
+        if (i + 1 < node->numOutputStream()) {
+            ss << ',';
+        }
+    }
+    ss << "],input:[";
+
+    for (int i = 0; i < node->numInputPin(); ++i) {
+        NamedPin *pin = node->inputPin(i);
+        ss << "{name:" << EscapeJson(pin->name())
+          << ",type:" << EscapeJson(pin->typeName());
+        if (pin->isConnected()) {
+            ss << ",connection:{node:"
+              << EscapeJson(pin->connectedStream()->node()->name())
+              << ",stream:" << EscapeJson(pin->connectedStream()->streamName())
+              << "}";
+        }
+        ss << "}";
+        if (i + 1 < node->numInputPin()) {
+            ss << ',';
+        }
+    }
+    ss << "]}";
+    reply->text += ss.str();
+}
+
+// Handle URI with prefix /node/nodeName, assuming nodeName is valid.
+void ServeValidNodeDir(std::vector<std::string> directories, NodeBase *node, HttpReply *reply) {
+
+    if (directories.size() == 2) {
+        // /node/<node name>, we list pins and streams.
+        ServeNode(node, reply);
+    } else {
+        if (directories[2] == "props") {
+            // /node/<node name>/props, we list node properties.
+            ListProperties(node, reply);
+        } else if (directories[2] == "stream") {
+            if (directories.size() >= 4) {
+                // /node/<node name>/stream/<stream name>
+                // we list stream properties
+                ListProperties(node->getOutputStreamByName(directories[3]), reply);
+            } else {
+                reply->text = "No stream name given\r\n";
+                reply->setNotFound();
+            }
+        } else if (directories[2] == "pin") {
+            if (directories.size() >= 4) {
+                // /node/<node name>/pin/<pin name>
+                // we list pin properties.
+                ListProperties(node->getInputPinByName(directories[3]), reply);
+            } else {
+                reply->text = "No pin name given\r\n";
+                reply->setNotFound();
+            }
+        } else {
+            reply->setNotFound();
+        }
+    }
+}
+
+// Handle URI with prefix: /node
+void ServeNodeDir(std::vector<std::string> directories, Graph *graph, HttpReply *reply) {
```

Comment:
same pass by ref here (and can be const)

Reply:
done